bib/cmd/bootc-image-builder/main.go

```diff
@@ -635,6 +634,17 @@ func run() error {
 	}
 	buildCmd.MarkFlagsRequiredTogether("aws-region", "aws-bucket", "aws-ami-name")
 
+	// If no subcommand is given, assume the user wants to use the build subcommand
+	// See https://github.com/spf13/cobra/issues/823#issuecomment-870027246
+	// which cannot be used verbatim because the arguments for "build" like
+	// "quay.io" will create an "err != nil". Ideally we could check err
+	// for something like cobra.UnknownCommandError but cobra just gives
+	// us an error string
+	_, _, err := rootCmd.Find(os.Args[1:])
+	if err != nil && !slices.Contains([]string{"help", "completion"}, os.Args[1]) {
+		args := append([]string{buildCmd.Use}, os.Args[1:]...)
```

Comment:
I think this should be `buildCmd.Name()`, which is the first word in the `Use` line.
The general form of `Use` is the full usage instructions of the command, including optional and position argument hints [1].  We don't use that and we only include the name of the command, but let's not have to deal with that if we ever extend the usage lines.

[`buildCmd.Name()`](https://pkg.go.dev/github.com/spf13/cobra#Command.Name) instead will always return the first word, even if we extend it.

https://pkg.go.dev/github.com/spf13/cobra#Command

Reply:
Thanks, nice catch! Updated